pkg/syncer/cnsoperator/controller/cnsnodevmbatchattachment/cnsnodevmbatchattachment_helper.go

```diff
@@ -506,12 +508,145 @@ func listAttachedFcdsForVM(ctx context.Context,
 	return attachedFCDs, nil
 }
 
+// convertDiskTypeToBackingType converts the diskType returned by QueryVirtualDiskInfo
+// to the appropriate CnsVolumeBackingType string used for batch attach operations.
+// The returned values correspond to VirtualDevice.FileBackingInfo subclasses as defined in
+// github.com/vmware/govmomi/cns/types (CnsVolumeBackingType constants).
+func convertDiskTypeToBackingType(diskType string) string {
```

Comment:
We should get this mapping verified from someone. It doesn't have all 7 types here.